sdk/attestation/azure-security-attestation/azure/security/attestation/_client.py

```diff
@@ -22,107 +17,151 @@
-from ._models import (
-    AttestationSigner,
-    AttestationToken,
-    AttestationResponse,
-    AttestationResult,
-    AttestationData,
-    TpmAttestationRequest,
-    TpmAttestationResponse)
-import base64
-from azure.core.tracing.decorator import distributed_trace
-from threading import Lock
+from ._models import AttestationSigner, AttestationToken, AttestationResult
+from ._common import merge_validation_args
+
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from azure.core.credentials import TokenCredential
 
 
 class AttestationClient(object):
     """An AttestationClient object enables access to the Attestation family of APIs provided
       by the attestation service.
 
-    :param str instance_url: base url of the service
     :param credential: Credentials for the caller used to interact with the service.
     :type credential: :class:`~azure.core.credentials.TokenCredential`
-    :keyword pipeline: If omitted, the standard pipeline is used.
-    :paramtype pipeline: Pipeline
-    :keyword policies: If omitted, the standard pipeline is used.
-    :paramtype policies: list[HTTPPolicy]
-    :keyword HttpTransport transport: If omitted, the standard pipeline is used.
-    :paramtype transport: HttpTransport
+    :param str endpoint: The attestation instance base URI, for example https://mytenant.attest.azure.net.
+    :keyword bool validate_token: if True, validate the token, otherwise return the token unvalidated.
+    :keyword validation_callback: Function callback to allow clients to perform custom validation of the token.
+        if the token is invalid, the `validation_callback` function should throw
+        an exception.
+    :paramtype validation_callback: ~typing.Callable[[AttestationToken, AttestationSigner], None]
+    :keyword bool validate_signature: if True, validate the signature of the token being validated.
+    :keyword bool validate_expiration: If True, validate the expiration time of the token being validated.
+    :keyword str issuer: Expected issuer, used if validate_issuer is true.
+    :keyword float validation_slack: Slack time for validation - tolerance applied
+        to help account for clock drift between the issuer and the current machine.
+    :keyword bool validate_issuer: If True, validate that the issuer of the token matches the expected issuer.
+    :keyword bool validate_not_before_time: If true, validate the "Not Before" time in the token.
+
+    .. tip::
+        The `validate_token`, `validation_callback`, `validate_signature`,
+        `validate_expiration`, `validate_not_before_time`, `validate_issuer`, and
+        `issuer` keyword arguments are default values applied to each API call within
+        the :py:class:`AttestationClient` class. These values can be
+        overridden on individual API calls as needed.
+
 
     For additional client creation configuration options, please see https://aka.ms/azsdk/python/options.
```

Comment:
For consistency with other links to this page:
```suggestion
    For additional client creation configuration options, please see `Python Request Options
    <https://aka.ms/azsdk/python/options>`_.
```